modules/oc-mirror-migration-process.adoc

```diff
@@ -0,0 +1,133 @@
+
+Migrating from oc-mirror v1 to v2 requires manually updating the ImageSetConfiguration, adjusting mirroring commands, and ensuring proper cleanup of v1 artifacts. Follow these steps to transition smoothly:
+
+. Modify the API version and remove deprecated fields in your `ImageSetConfiguration`. Refer below for the examples before and after the configuration:
++
+.Example output with oc-mirror plugin v1 configuration:
+[source,yaml]
+----
+kind: ImageSetConfiguration
+apiVersion: mirror.openshift.io/v1alpha2
+mirror:
+  platform:
+    channels:
+      - name: stable-4.17
+    graph: true
+  helm:
+    repositories:
+      - name: sbo
+        url: https://redhat-developer.github.io/service-binding-operator-helm-chart/
+  additionalImages:
+    - name: registry.redhat.io/ubi8/ubi:latest
+    - name: quay.io/openshifttest/hello-openshift@sha256:example_hash
+  operators:
+    - catalog: oci:///test/redhat-operator-index
+      packages:
+        - name: aws-load-balancer-operator
+storageConfig:  # REMOVE this field in v2
+  local:
+    path: /var/lib/oc-mirror
+----
++
+.Example output with oc-mirror plugin v2 configuration:
+[source,yaml]
+----
+kind: ImageSetConfiguration
+apiVersion: mirror.openshift.io/v2alpha1
+mirror:
+  platform:
+    channels:
+      - name: stable-4.17
+    graph: true
+  helm:
+    repositories:
+      - name: sbo
+        url: https://redhat-developer.github.io/service-binding-operator-helm-chart/
+  additionalImages:
+    - name: registry.redhat.io/ubi8/ubi:latest
+    - name: quay.io/openshifttest/hello-openshift@sha256:example_hash
+  operators:
+    - catalog: oci:///test/redhat-operator-index
+      packages:
+        - name: aws-load-balancer-operator
+----
+
+. Commands have changed to align with oc-mirror plugin v2's new workflow.
+
+.. For mirror-to-disk, run the following command:
++
+[source,terminal]
+----
+$ oc-mirror --config isc.yaml file://some-dir --v2
+----
+
+.. For disk-to-mirror, run the following command:
++
+[source,terminal]
+----
+$ oc-mirror --config isc.yaml --from file://some-dir docker://remote-registry --v2
+----
+
+.. For mirror-to-mirror, run the following command:
++
+[source,terminal]
+----
+$ oc-mirror --config isc.yaml --from file://some-dir docker://remote-registry --v2
+----
++
+[NOTE]
+====
+`--workspace` is now required for mirror-to-mirror operation.
+====
+
+. In v2, mirrored content is stored in different locations. You must check the cluster-resources directory inside their working directory for IDMS, ITMS, `CatalogSource`, and `ClusterCatalog` using the following command :
++
+[source,terminal]
+----
+$ ls <workspace>/cluster-resources/
+----
+
+. oc-mirror plugin v2 replaces ICSP with IDMS/ITMS. Before applying the new configuration, delete the outdated ICSP entries to avoid conflicts using the following command:
++
+[source,terminal]
+----
+$ oc delete imagecontentsourcepolicy --all
+----
+
+. Remove images mirrored using oc-mirror plugin v1 using the following command:
```

Comment:
This is more of an optional step.
We could introduce it by saying that v2 doesn' t auto-prune the destination registry, the way v1 does.
So we could tell users who are interrested in pruning images that are no longer needed to refer to the delete functionality of v2, with the reminder to use --delete-v1-images for more compatibility. 
We could say that once all v1 mirrored images have been deleted, the flag is no longer needed.
To be more precise, if they want to delete images mirrored with v2, they shouldn' t be setting that flag.

Reply:
The workflow to remove images mirrored by oc-mirror v1 should be : 
1) mirror the image by oc-mirror v1 (what error the mirror2disk+disk2mirror / mirror2mirror);
2) The ImageSetConfiguration API version changes from v1alpha2 (v1) to v2alpha1 (v2) , then run the v2 plugin with command : `oc-mirror -c isc.yaml file://anypath --v2`
3) ` oc-mirror delete --config=delete-isc.yaml --generate --delete-v1-images --workspace file://the-same-path-as-step2  docker://remote-registry --v2`
4) `oc-mirror delete --delete-yaml-file  the-same-path-as-step2/working-dir/delete/delete-images.yaml docker://remote-registry --v2`